src/main/java/com/vg/hm/samples/usermanagementsvc/service/impl/UserServiceImpl.java

```diff
@@ -64,4 +77,33 @@ public void updateUser(User user) {
         log.info("Listing all users!");
         return storage.values().stream().collect(Collectors.toList());
     }
+
+    @HystrixCommand(fallbackMethod = "fallback_hello", commandProperties = {@HystrixProperty(name="execution.isolation.thread.timeoutInMilliseconds", value = "1000")})
+     public String hello()throws InterruptedException{
+      Thread.sleep(3000);
+       return "Welcome Hystrix";
+     }
+     public String fallback_hello(){
+        return "Request fails. It takes long time to response";
+    }
+
+    @Override
+    public List<Make>getMakes() throws IOException {
+        String BASE_URL = "https://vpic.nhtsa.dot.gov/api/vehicles/";
+        Gson gson = new GsonBuilder()
+                .setDateFormat("yyy-MM-dd'T'HH:mm:ssz")
+                .create();
+
+        Retrofit retrofit = new Retrofit.Builder()
```

Comment:
You should remove this logic out of this service, and define a spring Config class where you can initialize this Bean using @Bean(name="vehicleServiceRetrofitClient"). This way you can inject that bean here like:
@Autowired
private VehicleAPIService retrofitClient;

Then you can later do this:
Call<MakesJson> call = retrofitClient.getAllMakes();

Cause right now, every single time you call this method, a new retrofit client gets created, and every single request has the same base url.

Reply:
OK! done